tasks/us/census/lodes.py

```diff
@@ -468,148 +466,76 @@ def columns(self):
             col.tags.append(source)
             col.tags.append(license)
         return cols
 
 
-class DownloadWorkplaceAreaCharacteristics(Task):
-    '''
-    Download all WAC files available
-    '''
-
+class DownloadUnzipLodes(DownloadGUnzipTask):
     year = IntParameter(default=2013)
+    filetype = Parameter(default='rac')
+    state = Parameter()
+    partorsegment = Parameter(default='S000')
+    jobtype = Parameter(default='JT00')
 
-    def requires(self):
-        for state in STATES - MISSING_STATES.get(self.year, set()):
-            yield DownloadLODESFile(filetype='wac',
-                                    part_or_segment='S000', # all jobs
-                                    year=self.year,
-                                    state=state)
+    def filename_lodes(self):
+        #   [STATE]_[FILETYPE]_[PART/SEG]_[TYPE]_[YEAR].csv.gz   where
+        return '{}_{}_{}_{}_{}.csv.gz'.format(self.state, self.filetype,
+                                              self.partorsegment,
+                                              self.jobtype, self.year)
 
-    def output(self):
-        for outfile in self.input():
-            yield outfile
+    def url(self):
+        return 'http://lehd.ces.census.gov/data/lodes/LODES7/{}/{}/{}'.format(
+            self.state, self.filetype, self.filename_lodes())
+
+    def download(self):
+        shell('wget -O {output}.gz {url}'.format(output=self.output().path, url=self.url()))
 
 
-class DownloadResidenceAreaCharacteristics(Task):
+class WorkplaceAreaCharacteristicsTemp(CSV2TempTableTask):
 
     year = IntParameter(default=2013)
 
-    def requires(self):
-        for state in STATES - MISSING_STATES.get(self.year, set()):
-            yield DownloadLODESFile(filetype='rac', year=self.year,
-                                    state=state, part_or_segment='S000') # all jobs
+    FILE_EXTENSION = 'csv'
 
-    def output(self):
-        for outfile in self.input():
-            yield outfile
+    def requires(self):
+        return [DownloadUnzipLodes(file_extension=self.FILE_EXTENSION,
+                                   filetype='rac', year=self.year,
+                                   state=state, partorsegment='S000',
+                                   jobtype='JT00') for state in STATES - MISSING_STATES.get(self.year, set())]
+
+    def input_csv(self):
+        csvs = []
+        for folder in self.input():
+            for file in os.listdir(folder.path):
+                if file.endswith('.{}'.format(self.FILE_EXTENSION)):
+                    csvs.append(os.path.join(folder.path, file))
+        return csvs
+
+    def coldef(self):
+        '''
+        We are receiving a list of files so we need to implement the coldef method
+        We are assuming that all the files have the same format (columns).
+        '''
+        csv = self.input_csv()[0]
+        header_row = shell('head -n 1 "{csv}"'.format(csv=csv), encoding=self.encoding).strip()
```

Comment:
Please use the CSVReader class to get the headers instead of using an external shell and we should remove it from CSV2TempTask class too but that is optional

Reply:
Solved for both lodes and `CSV2TempTableTask`